### PR TITLE
Move track_data_ to TimerTrack

### PR DESCRIPTION
**Pull request by dimitry-**
Only timer tracks use and store timer infos.

Test: builds, run Orbit take capture.

### CODE REVIEW

File: src/OrbitGl/Track.h [hunk 1]
```diff
@@ -126,7 +125,7 @@ class Track : public orbit_gl::CaptureViewElement, public std::enable_shared_fro
   bool draw_background_ = true;
   bool visible_ = true;
   bool pinned_ = false;
-  std::shared_ptr<orbit_client_data::TrackData> track_data_;
+  //  std::shared_ptr<orbit_client_data::TrackData> track_data_;
```
Comment (vickyliu-go4it) [inline src/OrbitGl/Track.h:128]: Maybe this can be removed now.

File: src/OrbitGl/PageFaultsTrack.h [hunk 2]
```diff
@@ -46,11 +46,16 @@ class PageFaultsTrack : public Track {
       uint64_t timestamp_ns, const std::array<double, kBasicPageFaultsTrackDimension>& values) {
     major_page_faults_track_->AddValuesAndUpdateAnnotations(timestamp_ns, values);
   }
+
   void AddValuesAndUpdateAnnotationsForMinorPageFaultsSubtrack(
       uint64_t timestamp_ns, const std::array<double, kBasicPageFaultsTrackDimension>& values) {
     minor_page_faults_track_->AddValuesAndUpdateAnnotations(timestamp_ns, values);
   }
 
+  uint32_t GetNumTimers() const override;
```
Comment (vickyliu-go4it) [inline src/OrbitGl/PageFaultsTrack.h:55]: `[[nodiscard]]` here and below

File: src/OrbitGl/TimerTrack.h [hunk 3]
```diff
@@ -110,6 +110,14 @@ class TimerTrack : public Track {
 
   float GetHeight() const override;
 
+  [[nodiscard]] std::vector<orbit_client_data::TimerChain*> GetChains() {
+    return track_data_->GetChains();
+  }
+
+  uint32_t GetNumTimers() const override;
```
Comment (vickyliu-go4it) [inline src/OrbitGl/TimerTrack.h:117]: `[[nodiscard]]` here and below

File: src/OrbitGl/GraphTrack.h [hunk 4]
```diff
@@ -45,6 +45,11 @@ class GraphTrack : public Track {
   virtual void AddValues(uint64_t timestamp_ns, const std::array<double, Dimension>& values) {
     series_.AddValues(timestamp_ns, values);
   }
+
+  uint32_t GetNumTimers() const override;
```
Comment (vickyliu-go4it) [inline src/OrbitGl/GraphTrack.h:49]: `[[nodiscard]]` for this and the following `Get*` methods

File: src/OrbitGl/PageFaultsTrack.cpp [hunk 5]
```diff
@@ -174,8 +174,17 @@ void PageFaultsTrack::OnTimer(const orbit_client_protos::TimerInfo& timer_info)
         static_cast<double>(system_page_faults - system_major_page_faults);
     AddValuesAndUpdateAnnotationsForMinorPageFaultsSubtrack(timer_info.start(), values);
   }
+}
+uint32_t PageFaultsTrack::GetNumTimers() const {
```
Comment (vickyliu-go4it) [inline src/OrbitGl/PageFaultsTrack.cpp:178]: I'm a little bit confused. From the name,  the`GetNumTimers()` method supposes to return the number of timers. For the `PageFaultsTrack`, we extract time series values for the major and minor subtracks from the same page faults timer (See `PageFaultsTrack::OnTimer`) for more details. Hence here we count the number of timers in the `PageFaultsTrack` twice.

Comment (vickyliu-go4it) [reply src/OrbitGl/PageFaultsTrack.cpp:178]: If we are counting the sum of graph track size here,  maybe we can add comments for the `Track::GetNumTimers()` method to explain that we are collecting different information for the `TimerTrack` and `GraphTrack`.

**Review by vickyliu-go4it — APPROVED** [review]
Thanks for making this change! LGTM otherwise some minor suggestions.

File: src/OrbitGl/GraphTrack.cpp [hunk 6]
```diff
@@ -102,11 +102,6 @@ void GraphTrack<Dimension>::UpdatePrimitives(Batcher* batcher, uint64_t min_tick
   DrawSeries(batcher, min_tick, max_tick, graph_z);
 }
 
-template <size_t Dimension>
-void GraphTrack<Dimension>::OnTimer(const orbit_client_protos::TimerInfo& timer_info) {
-  track_data_->AddTimer(/*depth=*/0, timer_info);
```
Comment (karupayun) [inline src/OrbitGl/GraphTrack.cpp:107]: I supposed that we were using OnTimer in GraphTrack

Comment (vickyliu-go4it) [reply src/OrbitGl/GraphTrack.cpp:107]: Suppose to be fine as `OnTimer` methods are override in the derived classes.

But I'm curious how the graph track related timers are stored in the capture file now.

**Review by karupayun — APPROVED** [review]

File: src/OrbitGl/PageFaultsTrack.h (hunk 2 above)
Comment (dimitry-) [reply src/OrbitGl/PageFaultsTrack.h:55]: done.

File: src/OrbitGl/GraphTrack.cpp (hunk 6 above)
Comment (dimitry-) [reply src/OrbitGl/GraphTrack.cpp:107]: we do not store TimerInfos in capture file anymore - only the raw data coming from the service

File: src/OrbitGl/GraphTrack.h (hunk 4 above)
Comment (dimitry-) [reply src/OrbitGl/GraphTrack.h]: done.

File: src/OrbitGl/PageFaultsTrack.cpp (hunk 5 above)
Comment (dimitry-) [reply src/OrbitGl/PageFaultsTrack.cpp]: ok, I looked closer and it appears we do not need GetNumTimers outside of ThreadTrack where they are used to sort tracks. I removed this from the base class and from here.

File: src/OrbitGl/TimerTrack.h (hunk 3 above)
Comment (dimitry-) [reply src/OrbitGl/TimerTrack.h]: done.

File: src/OrbitGl/Track.h (hunk 1 above)
Comment (dimitry-) [reply src/OrbitGl/Track.h]: removed.

File: src/OrbitGl/GraphTrack.cpp (hunk 6 above)
Comment (dimitry-) [reply src/OrbitGl/GraphTrack.cpp:107]: so all timer infos UI sees are calculated when orbit loads raw data from the file.

**Pull request merged**